docs/modules/ROOT/pages/glossary/index.adoc

```diff
@@ -1 +1,98 @@
 = Glossary of terms related to {ProductName}
+:icons: font
+:source-highlighter: highlightjs
+
+**{ProductName}** +
+A platform to automate the process of building, testing, and releasing applications. {ProductName} offers enterprise-grade security and customizable feature sets.   
+
+**build pipeline customization** +
+The ability to update and manage build pipelines for each component in an application. 
+
+**cluster** +
+A Kubernetes deployment with nodes that run containerized applications and a control plane that manages the nodes.
+
+**commit** +
+A change to one or more files. In {ProductName}, commits that you make to a linked git repository move through the pipeline and automatically get published.
+
+**component** +
+An image that {ProductName} builds from source code in a repository. One or more components that run together form an application.
```

Comment:
It isn't necessarily an image. A Component is an internal representation of an artifact that is being built ... including the linking between the source git repository and the storage location in the OCI registry.

We are also not trying to necessarily say that they run together to form an application (we also don't define an application here).